### ISSUE TITLE
Incorrect escaping of singlequotes in view comments

### ISSUE THREAD

**Issue by lkorpalski-pgs** [Snowflake-Labs/terraform-provider-snowflake#1049]
**Provider Version** 0.35.0
**Terraform Version** 1.1.2
**Describe the bug**

When specifying a comment in a snowflake_view resource which includes a single quote character ('), this doesn't get escaped properly, resulting in a semi-broken SQL, which is getting repeatedly replaced on each terraform run.

**Expected behavior**

Escape the single quote, or at least detect it as an illegal character and fail.

**Code samples and commands**
```
resource snowflake_view view {
  database = "db"
  schema   = "schema"
  name     = "view"

  comment = "comment with 'quotes' is broken"

  statement  = <<-SQL
    select * from foo;
SQL
  or_replace = false
  is_secure  = false
}
```

**Additional context**

This is a bug of the annoying type, as even with the incorrect SQL structure, the view is working, just tf produces additional output logs and recreates the resource repeatedly.

**Comment by monti-python** [comment]
This is still happening in version `0.83.0` and with different types of resources. We are unable to create a role with single quotes within the `comment` attribute, e.g:

```terraform
resource "snowflake_role" "schema_usage" {
  name = "MKT_PROD"
  comment = "Role that grants access on the 'MKT' schema in environment 'PROD'"
}
```

This fails with the error:

```text
│ Error: 001003 (42000): SQL compilation error:
│ syntax error line 1 at position 125 unexpected '' schema in environment ''.
```

**Comment by sfc-gh-jcieslak** [comment]
Hey @lkorpalski-pgs 
It should be the same case as in https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2411. Lucky there's an open pr (https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2448) that uses new SDK in view and materialized view, so after we merge and release it, the error shouldn't be there.

**Comment by sfc-gh-jcieslak** [comment]
Hey @lkorpalski-pgs  👋
Yesterday, we released a new provider version ([v0.95.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.95.0)) that introduces a lot of changes (also the previously mentioned pr should resolve this issue). Please upgrade using the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md). Closing this one, please create another issue if the problem still persists in the newer versions.

**Comment by lkorpalski-pgs** [comment]
Unfortunately I can't confirm the fix myself, as I got moved to another project since then. Thank you for the notice though.
